internal/pkg/image/driver/squashfuse/driver.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Contributors to the Apptainer project, established as
+//   Apptainer a Series of LF Projects LLC.
+//   For website terms of use, trademark policy, privacy policy and other
+//   project policies see https://lfprojects.org/policies
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package squashfuse
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/apptainer/apptainer/pkg/image"
+	"github.com/apptainer/apptainer/pkg/sylog"
+	"github.com/apptainer/apptainer/pkg/util/apptainerconf"
+	"github.com/apptainer/apptainer/pkg/util/fs/proc"
+	"github.com/syndtr/gocapability/capability"
+)
+
+const driverName = "squashfuse"
+
+type squashfuseDriver struct {
+	unprivileged bool
+	cmd          *exec.Cmd
+}
+
+func Register(unprivileged bool, fileconf *apptainerconf.File) error {
+	if fileconf.ImageDriver != "" {
```

Comment:
you also need to change the logic to `if fileconf.ImageDriver != "squashfsfuse"`

Reply:
The point of that is to allow a configured driver to override the internal one, so I think it is correct as it is.  I'll add a comment to say what it is for.